src/pages/SettingsPage/components/SettingsHelperContent/index.js

```diff
@@ -1,34 +1,61 @@
-import React from 'react'
-
-import API from 'api'
+import React, { useEffect, useState } from 'react'
+import { useSelector, useDispatch } from 'react-redux'
 
 import * as Styles from './styles'
+import {
+  fetchSettings,
+  putSettings,
+  selectSettingsData,
+  selectSettingsStatus,
+} from 'features/settings/settingsSlice'
+import { dataLoadingStatus } from 'hooks/dataLoadingStatus'
 
 const SettingsHelperContent = () => {
-  const useSettings = API.useSettings()
-  const [mutate_PUT_Settings] = API.usePutSettings()
+  const dispatch = useDispatch()
+  const [updateSettings, setUpdateSettings] = useState(false)
+
+  useEffect(() => {
+    dispatch(fetchSettings())
+  }, [dispatch, updateSettings])
```

Comment:
pusta tablica w zaleznosciach. Do useEffect daje się tylko dane, które ulegną zmianie, tak zeby wywolac go jeszcze raz. Więc dispatch nie, a co do updateSettings:
Wywal 15 linijkę (z updateSettings) całkiem, a w handleputSettings po prostu odpal ponownie dispatch(fetchSettings())

Reply:
okej, a przy dispatch gdy go nie dodaje wyskakuje komunikat w konsoli : React Hook useEffect has a missing dependency: 'dispatch'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

Reply:
dodaj do eslinta regułę
"react-hooks/exhaustive-deps": "warn"

Reply:
Generalnie to ta zasada powinna być zachowana, ale w przypadku dispatch to nie ma sensu, bo dispatch się nie zmienia. Wolę nie mieć warningów, a sam martwić się o dokładanie do zależności używanych funkcji (bo czasem są przypadki, że używasz jakiejś, a nie chcesz na nią reagować w useEffect)

Reply:
Rozumiem ale mimo wszystko puki co chciałbym mieć warningi więc zostawię dispatcha w tablicy.
Pomijam.

Reply:
Skipped.